Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest/Contest.tsx

```diff
@@ -98,34 +103,67 @@ const Contest = () => {
         [],
     );
 
-    return (
-        <>
-            <div className={styles.headingContest}>
-                <Heading
-                  type={HeadingType.primary}
-                  className={styles.contestHeading}
-                >
-                    {contest?.name}
-                </Heading>
-                <Heading type={HeadingType.secondary} className={secondaryHeadingClassName}>
-                    {renderTimeRemaining()}
-                    {renderScore()}
-                </Heading>
-            </div>
-
-            <div className={styles.contestWrapper}>
-                <div className={navigationContestClassName}>
-                    <ContestTasksNavigation />
+    const renderErrorMessage = useCallback(() => (
+        <div className={styles.headingContest}>
+            <Heading
+              type={HeadingType.primary}
+              className={styles.contestHeading}
+            >
+                {contest?.name}
+                -
+                {validationResult.message}
+            </Heading>
+        </div>
+    ), [ contest, validationResult ]);
+
+    const renderContest = useCallback(
+        () => (
+            <>
+                <div className={styles.headingContest}>
+                    <Heading
+                      type={HeadingType.primary}
+                      className={styles.contestHeading}
+                    >
+                        {contest?.name}
+                    </Heading>
+                    <Heading type={HeadingType.secondary} className={secondaryHeadingClassName}>
+                        {renderTimeRemaining()}
+                        {renderScore()}
+                    </Heading>
                 </div>
-                <div className={submissionBoxClassName}>
-                    <SubmissionBox />
-                </div>
-                <div className={problemInfoClassName}>
-                    <ContestProblemDetails />
+
+                <div className={styles.contestWrapper}>
+                    <div className={navigationContestClassName}>
+                        <ContestTasksNavigation />
+                    </div>
+                    <div className={submissionBoxClassName}>
+                        <SubmissionBox />
+                    </div>
+                    <div className={problemInfoClassName}>
+                        <ContestProblemDetails />
+                    </div>
                 </div>
-            </div>
-        </>
+            </>
+        ),
+        [
+            contest,
+            navigationContestClassName,
+            problemInfoClassName,
+            renderScore,
+            renderTimeRemaining,
+            secondaryHeadingClassName,
+            submissionBoxClassName,
+        ],
     );
+
+    const renderPage = useCallback(
+        () => validationResult.isValid
+            ? renderContest()
+            : renderErrorMessage(),
+        [ renderErrorMessage, renderContest, validationResult ],
+    );
```

Comment:
For once I like it more the way it is now, because to me it is more obvious what we are rendering at first glance without having to go into the other methods. Also, we are checking only once if the result is valid.

Reply:
In my opinion, both ways are readable and look fine to me. So far I am writing the logic the same as @msmner, but I am fine with changing to @Minkov's way.

Reply:
I find @msmner's way of writing this easier to read and agree with not needing to go inside the other methods to understand when and under what circumstances the functions render a component.